pkg/proc/mem.go

```diff
@@ -136,3 +136,18 @@ func DereferenceMemory(mem MemoryReadWriter) MemoryReadWriter {
 	}
 	return mem
 }
+
+// bufferMemoryReadWriter is dummy a MemoryReadWriter backed by a []byte.
+type bufferMemoryReadWriter struct {
```

Comment:
This is fine for now, but this makes me think this would be better conforming to the interfaces in the `io` package so we don't have to re-invent `bytes.Buffer`.